rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -317,25 +317,45 @@ unsigned rocalTensor::copy_data(hipStream_t stream, void *host_memory, bool sync
 }
 #endif
 
-unsigned rocalTensor::copy_data(void *user_buffer) {
+unsigned Tensor::copy_data(void *user_buffer, RocalOutputMemType external_mem_type) {
     if (_mem_handle == nullptr) return 0;
 
+    if(external_mem_type == RocalOutputMemType::ROCAL_MEMCPY_GPU) {
 #if ENABLE_HIP
-    if (_info._mem_type == RocalMemType::HIP) {
-        // copy from device to host
-        hipError_t status;
-        if ((status = hipMemcpyDtoH((void *)user_buffer, _mem_handle, _info.data_size())))
-            THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
-    } else
+        if (_info._mem_type == RocalMemType::HIP) {
+            // copy from device to device
+            hipError_t status;
+            if ((status = hipMemcpyDtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
```

Comment:
@fiona-gladwin Error thrown is wrong. Change to ```hipMmecpyDtoD```

Reply:
Done